service/history/workflow/mutable_state_impl.go

```diff
@@ -3024,18 +3024,10 @@ func (ms *MutableStateImpl) ApplyBuildIdRedirect(
 	// cluster and events applied by WF rebuilder.
 	ms.GetExecutionInfo().BuildIdRedirectCounter = redirectCounter
 
-	// Re-scheduling pending WF and activity tasks which are not started yet.
-	if ms.HasPendingWorkflowTask() && !ms.HasStartedWorkflowTask() &&
-		ms.GetPendingWorkflowTask().ScheduledEventID != startingTaskScheduledEventId &&
-		// TODO: something special may need to be done for speculative tasks as GenerateScheduleWorkflowTaskTasks
-		// does not support them.
-		ms.GetPendingWorkflowTask().Type != enumsspb.WORKFLOW_TASK_TYPE_SPECULATIVE {
-		// sticky queue should be cleared by UpdateBuildIdAssignment already, so the following call only generates
-		// a WorkflowTask, not a WorkflowTaskTimeoutTask.
-		err = ms.taskGenerator.GenerateScheduleWorkflowTaskTasks(ms.GetPendingWorkflowTask().ScheduledEventID)
-		if err != nil {
-			return err
-		}
+	// Re-scheduling pending workflow and activity tasks.
+	err = ms.reschedulePendingWorkflowTask(false)
```

Comment:
This seems like a case where we want to invalidate existing tasks? i.e `ms.reschedulePendingWorkflowTask(true)`?